docs/docs/gestures/gesture-detector.md

```diff
@@ -47,3 +47,25 @@ This parameter allows to specify which `userSelect` property should be applied t
 - Gesture Detector will use first native view in its subtree to recognize gestures, however if this view is used only to group its children it may get automatically [collapsed](https://reactnative.dev/docs/view#collapsable-android). Consider this example:
   <FunctionalComponents />
   If we were to remove the collapsable prop from the View, the gesture would stop working because it would be attached to a view that is not present in the view hierarchy. Gesture Detector adds this prop automatically to its direct child but it's impossible to do automatically for more complex view trees.
+
+- Using the same instance of gesture handler across multiple Gesture Detectors is not possible. Have a look at the code below:
+
+  ```jsx
+  export default function Example() {
+    const pan = Gesture.Pan();
+
+    return (
+      <View>
+        <GestureDetector gesture={pan}>
+          <View>
+            <GestureDetector gesture={pan}>
+              <View />
+            </GestureDetector>
+          </View>
+        </GestureDetector>
+      </View>
+    );
+  }
+  ```
+
+  This example will throw an error, becuse we try to use the same instance of `Pan` in two different Gesture Detectors.
```

Comment:
I'm not sure whether we need an example for this, cc. @kacperkapusciak though.

Maybe it would be worth mentioning that composition is not the way and `.*ExternalGesture` methods should be used in cases like that.

Reply:
Programmers tend to skim the docs and only brifely read the code snippets so a snippet here is fine.

But please add a `// don't do this!` comment to show that it won't work

@m-bert could you add that?

Reply:
Of course! Done in [d6f34d6](https://github.com/software-mansion/react-native-gesture-handler/pull/2694/commits/d6f34d65e1df939c3a492fc5e317090e8c1e83bc).